autoscoper/src/net/Socket.cpp

```diff
@@ -262,15 +271,14 @@ void Socket::createNewConnection()
 {
   std::cerr << "New Matlab Client is Connected..." << std::endl;
   QTcpSocket *clientConnection = tcpServer->nextPendingConnection();
-  connect(clientConnection, &QAbstractSocket::disconnected, this, &Socket::deleteConnection);
```

Comment:
Existing workflow not explicitly closing the connection will behave slightly differently. Do we anticipate to be an issue ?

Reply:
Added this back and properly handled the error in the `deleteConnection` function.